### ISSUE TITLE
Error in Success callbackId facing issue ios

### ISSUE THREAD

**Issue by sksshareef** [arielfaur/ionic-audio#26]
Hi I am facing this error,
in my project in one page I am getting all audio list dynamically and next page with controller 
calling the audio like this 
$scope.track = $rootScope.songsObject[$stateParams.id];
and the audio is playing well no issue I am going back again getting the list of audio's and again selected another one its working but same after 3 to 4 audio's just the spinner is rotating no errors in Xcode also and nothing is working I have youtube videos also they are also not working at that time I think in background some thing is happening and some times i am getting this error in Xcode  
Error in Success callbackId: Media1035323214 : TypeError: 'undefined' is not an object (evaluating 'currentTrack.progress = position')
please help me

**Comment by sksshareef** [comment]
Its working fine in the simulater but facing the issue on iphone ios 7.1.2

**Comment by arielfaur** [comment]
Could you provide a code example? I cannot follow what you are trying to do.

**Comment by sksshareef** [comment]
this is my song list controller 

ionicApp.controller("SongController",['$scope','$state', '$http', 'ApiEndpoint','$rootScope','$ionicLoading', '$cordovaNetwork','$cordovaDialogs', function($scope, $state, $http, ApiEndpoint,$rootScope, $ionicLoading, $cordovaNetwork, $cordovaDialogs) {
  $ionicLoading.show({
    template: '<ion-spinner></ion-spinner>'
  })
	$rootScope.songsObject = null;
	$scope.songItems = [];
	$scope.loadSongs = function(){
	$http({
    method: 'POST',
    url: ApiEndpoint.url+"/songsGetios.php",
    data:"pageNo=1&perPage=100",
    headers: {'Content-Type': 'application/x-www-form-urlencoded'}
}).then(function(response) {
    console.log('Success', response.data.message);
	$scope.songItems = response.data.message;
	$rootScope.songsObject = $scope.songItems;
	$ionicLoading.hide();
    // For JSON responses, resp.data contains the result
  }, function(err) {
    console.error('ERR', err.status);
    $ionicLoading.hide();
    // err.status will contain the status code
    if(err.status === 0){
      $cordovaDialogs.alert('No Internet','Bathukamma','Ok');
    }
  })
}
}]);
and if we select any one song am passing the selected id and track URL and etc details 

this my player controller 

ionicApp.controller('MusicCtrl', ['$scope','$rootScope','$stateParams',function($scope, $rootScope, $stateParams) {
    $scope.tracks = null;
    $scope.currentTrack = null;
    $scope.track = $rootScope.songsObject[$stateParams.id];
    console.log($scope.track);
      $scope.$on('$ionicView.beforeLeave', function(){
        console.log('Track'+$scope.track);
    $scope.track = null;
  });
  $scope.$on('$ionicView.afterLeave', function(){
    console.log('Track'+$scope.track);
    $scope.track = null;
  });
}]);

this is my player html file 

<ion-view title="Songs">
	<ion-pane class ="contentBg">
    	<ion-content>
<img ng-src="img/songs_slider.jpg">
      <ion-audio-track  class="audio-spinner" track="track">
        <div>
            <div class="item-audio-title"><img ng-src="{{track.teluguImage}}"></div>
        </div>   
        <div>
         <ion-audio-controls>
               <a class="button button-large icon" ion-audio-play></a>
               <ion-spinner class="spinner" icon="ios"></ion-spinner>     
        </ion-audio-controls>
        <ion-audio-progress-bar display-time></ion-audio-progress-bar>
</div>
     </ion-audio-track>
    	</ion-content>
    </ion-pane>
</ion-view>

media is playing perfectly, but the error case is 
when playing audio I will go back select another audio that's also playing but if the same done for more then 4 or 5 audio, the page spinner is just loading and I am getting that error in the Xcode console and in the same app I have youtube videos also which are also not playing after this event .

**Comment by arielfaur** [comment]
It's very hard to follow the code as it is. Maybe you can apply some formatting. Besides looking at the Xcode console you should open the Safari developer console, and then pick iPhone for remote debugging. There might be a Javascript error that you are not seeing. 
Have you done that? It is explained here:
http://developer.telerik.com/featured/a-concise-guide-to-remote-debugging-on-ios-android-and-windows-phone/

**Comment by sksshareef** [comment]
Thanks for quick reply bro ,

Ya i did the same and with some neat coding

when the song is playing am going back and selecting one more song and song
is playing again going back at that time i got some error in web console

Error in Success callbackId: Media642205264 : TypeError: 'undefined' is not
an object (evaluating 'currentTrack.progress = position')
TypeError: 'undefined' is not an object (evaluating 'currentTrack.progress
= position')

please find the attachment for clear error and event

On Fri, Dec 4, 2015 at 3:53 AM, arielfaur <notifications@github.com> wrote:

> It's very hard to follow the code as it is. Maybe you can apply some
> formatting. Besides looking at the Xcode console you should open the Safari
> developer console, and then pick iPhone for remote debugging. There might
> be a Javascript error that you are not seeing.
> Have you done that? It is explained here:
>
> http://developer.telerik.com/featured/a-concise-guide-to-remote-debugging-on-ios-android-and-windows-phone/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/arielfaur/ionic-audio/issues/26#issuecomment-161805119>
> .
>



-- 
Regards,
*Shaik Sharif** |** Web Developer*

**Comment by sksshareef** [comment]
![screen shot 2015-12-04 at 12 44 22 pm](https://cloud.githubusercontent.com/assets/14145809/11583953/68a22456-9a85-11e5-8a61-240ea3eb70f4.png)

**Comment by sksshareef** [comment]
After this error i am able to move around the app and others songs also but when ever i select others song i get the object and ionic-audio : playing track track-name also 
but on screen just spinner is rotating 
please help ,
thanks

**Comment by sksshareef** [comment]
And also my updated Controller Code

ionicApp.controller('MusicCtrl2', function($scope,$rootScope, $stateParams)
{
  $scope.song = $rootScope.songsObject[$stateParams.id];
  console.log($scope.song);
  var songUrl = $scope.song.url;
  var songTitle = $scope.song.title;
  var songImage = $scope.song.teluguImage;

    $scope.tracks = [
        {
            url: songUrl,
            artist: songTitle,
            title: songTitle,
            art: songImage
        }
    ];
});

On Fri, Dec 4, 2015 at 12:47 PM, Shaik Sharif <shareefleadwinner@gmail.com>
wrote:

> Thanks for quick reply bro ,
>
> Ya i did the same and with some neat coding
>
> when the song is playing am going back and selecting one more song and
> song is playing again going back at that time i got some error in web
> console
>
> Error in Success callbackId: Media642205264 : TypeError: 'undefined' is
> not an object (evaluating 'currentTrack.progress = position')
> TypeError: 'undefined' is not an object (evaluating 'currentTrack.progress
> = position')
>
> please find the attachment for clear error and event
>
> On Fri, Dec 4, 2015 at 3:53 AM, arielfaur <notifications@github.com>
> wrote:
>
>> It's very hard to follow the code as it is. Maybe you can apply some
>> formatting. Besides looking at the Xcode console you should open the Safari
>> developer console, and then pick iPhone for remote debugging. There might
>> be a Javascript error that you are not seeing.
>> Have you done that? It is explained here:
>>
>> http://developer.telerik.com/featured/a-concise-guide-to-remote-debugging-on-ios-android-and-windows-phone/
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/arielfaur/ionic-audio/issues/26#issuecomment-161805119>
>> .
>>
>
>
>
> --
> Regards,
> *Shaik Sharif** |** Web Developer*
>



-- 
Regards,
*Shaik Sharif** |** Web Developer*

**Comment by sksshareef** [comment]
one more use case ,
if before going back if i pause the song and go back and select other songs doing the same number of time , Then its working fine and great 
so i think may be the error is with media is not getting released properly when pressing back while  audio is playing .

**Comment by sksshareef** [comment]
Great work bro that problem is resolved and now the performance also greatly improved .
but one small issue when i click the play button spinner is rotating and getting hang till it play the file

**Comment by arielfaur** [comment]
That is the way Cordova Media plugin works in iOS it freezes the UI while loading the file.

**Comment by sksshareef** [comment]
Thankss so kuch bro great work  and looking forward more functionalities
